dataflow/src/main/java/org/checkerframework/dataflow/cfg/visualize/CFGVisualizeLauncher.java

```diff
@@ -267,12 +269,36 @@ public void write(int b) throws IOException {}
         return res.getCFG();
     }
 
+    /**
+     * Write down generated String representation of the CFG.
+     *
+     * @param analysis instance of forward or backward analysis from specific dataflow test case.
+     */
+    @SuppressWarnings("CatchAndPrintStackTrace") // we want to use e.printStackTrace here.
+    public static void writeDownStringOfCFG(Analysis<?, ?, ?> analysis) {
+
+        String inputFile = "Test.java";
+        String method = "test";
+        String clazz = "Test";
+        String outputFile = "Out.txt";
+
+        Map<String, Object> res = generateStringOfCFG(inputFile, method, clazz, true, analysis);
+        try (FileWriter out = new FileWriter(outputFile)) {
+            if (res != null && res.get("stringGraph") != null) {
+                out.write(res.get("stringGraph").toString());
+            }
+            out.write("\n");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
     /**
      * Invoke "dot" command to generate a PDF.
      *
      * @param file name of the dot file
      */
-    protected void producePDF(String file) {
+    public static void producePDF(String file) {
```

Comment:
I'm personally against changing these methods from `protected` to `public static`. One reason is they will look like global utils, but you only need them in this class. Could you explain the motivation behind these changes?

Reply:
> protected accessor will report warning and fail the test
Could you provide an example? I'm not sure why it reports warning and what warning will be reported?

Reply:
For example:
```
warning: [ProtectedMembersInFinalClass] Make members of final classes package-private: producePDF 
protected static void producePDF(String file) {
```
But sure I can make it private.

Reply:
Ah, didn't know there's such warning message. Making them private sounds good to me.

Reply:
Okay!